### PR TITLE
Oliver short dashboard

### PR DESCRIPTION
**Pull request by olliejth**
Continued work on dashboard, added botanist names to pie chart and adapted charts to work with real data.

### CODE REVIEW

File: dashboard/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Specifies latest image of python
+FROM python:latest
+
```
Comment (mehrdadhalali) [inline dashboard/Dockerfile:3]: Could have a WORKDIR

File: dashboard/load_dashboard.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+"""Loads visualisations to constantly running dashboard."""
+
+import pandas as pd
+import streamlit as st
+
+from extract_dashboard import extract_readings
+from transform_dashboard import create_botanist_pie, create_temperature_bar, create_moisture_bar, create_temperature_line
```
Comment (mehrdadhalali) [inline dashboard/load_dashboard.py]: If pylint yells at you, you can break this into multiple import lines

File: dashboard/load_dashboard.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    plants_df = extract_readings()
+
+    botanist_plants_pie = create_botanist_pie(plants_df)
+    temp_bar_chart = create_temperature_bar(plants_df)
+    temp_line_chart = create_temperature_line(plants_df)
+    moist_bar_chart = create_moisture_bar(plants_df)
+
+    plant_ids = plants_df["plant_id"].unique().tolist()
+    st.sidebar.markdown("# Sidebar")
+    st.sidebar.multiselect("Trucks", options=plant_ids)
```
Comment (Leon-JS02) [inline dashboard/load_dashboard.py]: Trucks?

File: dashboard/load_dashboard.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    st.altair_chart(botanist_plants_pie, use_container_width=True)
+    st.altair_chart(temp_bar_chart, use_container_width=True)
+    st.altair_chart(temp_line_chart, use_container_width=True)
+    st.altair_chart(moist_bar_chart, use_container_width=True)
+
+
+# cols = st.columns(2)
+#    with cols[0]:
+#         st.altair_chart(total_transactions, use_container_width=True)
```
Comment (Leon-JS02) [inline dashboard/load_dashboard.py]: Commented out code from truck dashboard?

File: dashboard/transform_dashboard.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    temp_line_chart = alt.Chart(df_filtered, title=title).mark_line().encode(
+        x='at:T',  # 'at' column for time on x-axis
+        y='temperature:Q',  # 'temperature' column for y-axis (quantitative)
+        color='plant_id:N'  # Different color for each plant_id
+    )
+
+    return temp_line_chart
+
+
+def create_moisture_bar(df: pd.DataFrame):
```
Comment (Leon-JS02) [inline dashboard/transform_dashboard.py]: I like these functions for creating the functions. They could be type hinted with `alt.Chart` as their return value, though!

**Review by Leon-JS02 — CHANGES_REQUESTED** [review]
Good and clean dashboarding code! A few small changes though.

File: dashboard/transform_dashboard.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    temp_bar_chart = alt.Chart(recent_df, title=title).mark_bar().encode(
+        x='plant_id:N',
+        y='temperature:Q'
+    )
+
+    return temp_bar_chart
+
+
+def create_temperature_line(df: pd.DataFrame):
+    """Creates bar chart of plants against their most recently recorded temperature."""
```
Comment (mehrdadhalali) [inline dashboard/transform_dashboard.py]: Same docstring as above

**Review by mehrdadhalali — APPROVED** [review]

**Review by Leon-JS02 — APPROVED** [review]

**Pull request merged**